interpreter/cling/lib/Interpreter/IncrementalParser.cpp

```diff
@@ -728,10 +728,20 @@ namespace cling {
     SourceLocation NewLoc = getLastMemoryBufferEndLoc().getLocWithOffset(1);
 
     llvm::MemoryBuffer* MBNonOwn = MB.get();
-    // Create FileID for the current buffer
-    FileID FID = SM.createFileID(std::move(MB), SrcMgr::C_User,
-                                 /*LoadedID*/0,
-                                 /*LoadedOffset*/0, NewLoc);
+
+    // Create FileEntry and FileID for the current buffer
+    const clang::FileEntry* FE
+       = SM.getFileManager().getVirtualFile("vfile for " + source_name.str(),
```

Comment:
Can we do this *only* if there is a completion point? This changes the filename and "file-ness" of all memory buffers...